Nagstamon/Config.py

```diff
@@ -179,20 +179,20 @@ def __init__(self):
 
         # Parse the command line
         parser = argparse.ArgumentParser(description='Nagios status monitor for your desktop')
-        parser.add_argument('cfgpath', nargs='*', help='Path for configuration folder')
+        parser.add_argument('cfgpath', nargs='?', help='Path for configuration folder')
+        parser.add_argument('-psn', action='store_true',
+            help='force ~/.nagstamon as config folder (used by launchd in MacOSX)')
         # necessary because otherwise setup.py goes crazy of argparse
         args, unknown = parser.parse_known_args()
 
         # try to use a given config file - there must be one given
         # if sys.argv is larger than 1
-        if args.cfgpath:
-            # MacOSX related -psn argument by launchd
-            if '-psn' in args.cfgpath:
-                # new configdir approach
-                self.configdir = os.path.expanduser('~') + os.sep + '.nagstamon'
-            else:
-                # allow to give a config file
-                self.configdir = args.cfgpath[0]
+        if args.psn:
```

Comment:
I am sorry I've got confused. I wrote this code this morning, how could it crash yesterday?
Anyway, take your time ;)

Thank you for your work.